services/elastic_ip_manager/src/main/java/com/futurewei/alcor/elasticipmanager/controller/ElasticIpController.java

```diff
@@ -0,0 +1,484 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+
+package com.futurewei.alcor.elasticipmanager.controller;
+
+import com.futurewei.alcor.common.entity.ResponseId;
+import com.futurewei.alcor.common.utils.Ipv4AddrUtil;
+import com.futurewei.alcor.common.utils.Ipv6AddrUtil;
+import com.futurewei.alcor.elasticipmanager.config.IpVersion;
+import com.futurewei.alcor.elasticipmanager.exception.ElasticIpIdConfilictException;
+import com.futurewei.alcor.elasticipmanager.exception.ElasticIpNoProjectIdException;
+import com.futurewei.alcor.elasticipmanager.exception.ElasticIpProjectIdConflictException;
+import com.futurewei.alcor.elasticipmanager.exception.elasticip.*;
+import com.futurewei.alcor.elasticipmanager.exception.ElasticIpQueryFormatException;
+import com.futurewei.alcor.elasticipmanager.exception.elasticiprange.ElasticIpRangeBadRangesException;
+import com.futurewei.alcor.elasticipmanager.exception.elasticiprange.ElasticIpRangeNoIdException;
+import com.futurewei.alcor.elasticipmanager.exception.elasticiprange.ElasticIpRangeVersionException;
+import com.futurewei.alcor.elasticipmanager.service.ElasticIpRangeService;
+import com.futurewei.alcor.elasticipmanager.service.ElasticIpService;
+import com.futurewei.alcor.web.entity.elasticip.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.*;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@RestController
+public class ElasticIpController {
+    private static final Logger LOG = LoggerFactory.getLogger(ElasticIpController.class);
+
+    @Autowired
+    ElasticIpService elasticipService;
+
+    @Autowired
+    ElasticIpRangeService elasticIpRangeService;
+
+    private boolean isIpAddressInvalid(String ipAddress, Integer ipVersion) {
```

Comment:
To make the controller method a bit explicit, I would recommend to create a controllerUtil class and move the private methods there.

Reply:
OK, will do

Reply:
Done